ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -114,6 +115,7 @@ public int compare(IndexEntry e1, IndexEntry e2) {
   private volatile ConcurrentSkipListMap<Offset, IndexSegment> inFluxIndexSegments = validIndexSegments;
   private final Logger logger = LoggerFactory.getLogger(getClass());
   private final IndexPersistor persistor = new IndexPersistor();
+  private ScheduledFuture<?> persistorTask = null;
```

Comment:
looks like it couldn't be `final` because it brings either `cannot assign a value to a final variable` or `variable might not have been initialized` compile error.

Reply:
Discussed offline - this should be possible